python_modules/dagster/dagster/_core/definitions/asset_condition/asset_condition_evaluation_context.py

```diff
@@ -55,6 +55,158 @@ def wrapped(self: Any) -> Any:
     return wrapped
 
 
+@dataclass(frozen=True)
+class NewAssetConditionEvaluationContext:
+    """This is a new version of the AssetConditionEvaluationContext that is significantly
+    simplified. Refactoring will need to be done to convert all existing uses of the old class,
+    so for now just keeping this as a separate class.
+    """
+
+    from ...asset_graph_view.asset_graph_view import AssetGraphView, AssetSlice
+
+    condition: "AssetCondition"
+    root_condition: "AssetCondition"
+
+    asset_graph_view: AssetGraphView
+    candidate_slice: AssetSlice
+
+    previous_evaluation: Optional["AssetConditionEvaluation"]
+    previous_evaluation_state: Optional["AssetConditionEvaluationState"]
+
+    evaluation_state_by_key: Mapping[AssetKey, "AssetConditionEvaluationState"]
+
```

Comment:
what will the meaning of these be outside of the asset daemon?

Reply:
previous_evaluation / previous_evaluation_state can be None inside the asset daemon (for the first evaluation of an asset), and outside the asset daemon, I'd imagine that these would always be None. either way, all AssetConditions will need to be able to function without a previous evaluation state.

the evaluation_state_by_key value will only be used inside the asset daemon, but likewise is sensible to be empty when evaluating outside the daemon. I'm still not sure the best factoring here, but in order for us to create runs that consist of multiple assets at once, we'll need some concept of what _will be_ materialized within the context of a run that we're building up within a single tick. The most verbose way of doing this would be to have separate AssetConditions for the "is currently" and "will be", e.g. `AssetCondition.parent_newer() | AssetCondition.parent_will_be_newer()`. The latter would always be False when evaluating outside of the Daemon context. However, I suspect that would be too verbose / confusing, and so we'd want to do something like we currently do (which is to combine both the "is" and "will be" into a single rule), or do something like:

```
ParentAssetCondition.newer().or_will_update()

ParentAssetCondition.updated_since_cron().or_will_update()
```

where ParentAssetCondition is a specific type of condition that has convenience properties (e.g. the ability to target it at specific parents).